apps/src/pythonlab/PythonlabView.tsx

```diff
@@ -169,13 +173,41 @@ const PythonlabView: React.FunctionComponent<
   // Set view code to false if level is switched for any levels in widget view.
   useLifecycleNotifier(LifecycleEvent.LevelLoadStarted, () => {
     dispatch(setWidgetViewShowCode(false));
+    setAiTutor2HintQuestion(undefined);
   });
 
+  // Given a question for the AITutor2, return the full question to ask, which means
+  // appending all the relevant context.
+  const getAiTutor2FullQuestionFromQuestion = (
+    question: string,
+    questionType: 'hint' | 'user'
+  ) => {
+    const fullQuestion = [
+      question,
+      'Here is my code:',
+      source.openFiles
```

Comment:
We probably want all files, not just the open ones, right?

Reply:
I've now made an attempt at enumerating all files.  Would appreciate feedback on the attempt at filtering that I borrowed from elsewhere.